From dcc8727870eddec05b801adec3ce6c557683e806 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Sun, 20 Feb 2022 12:10:37 +0000
Subject: [PATCH] jinja2: fix compatibility with python3-3.10

Use collection.abc to fix build with python3-3.10:

| FAILED: v8_snapshot_clang_x64/gen/v8/src/inspector/protocol/Forward.h v8_snapshot_clang_x64/gen/v8/src/inspector/protocol/Protocol.cpp v8_snapshot_clang_x64/gen/v8/src/ins
pector/protocol/Protocol.h v8_snapshot_clang_x64/gen/v8/src/inspector/protocol/Console.cpp v8_snapshot_clang_x64/gen/v8/src/inspector/protocol/Console.h v8_snapshot_clang_x64/gen/v8/src/inspector/protocol/Debugger.cpp v8_snapshot_clang_x64/gen/v8/src/inspector/protocol/Debugger.h v8_snapshot_clang_x64/gen/v8/src/inspector/protocol/HeapProfiler.cpp v8_snapshot_clang_x64/gen/v8/src/inspector/protocol/HeapProfiler.h v8_snapshot_clang_x64/gen/v8/src/inspector/protocol/Profiler.cpp v8_snapshot_clang_x64/gen/v8/src/inspector/protocol/Profiler.h v8_snapshot_clang_x64/gen/v8/src/inspector/protocol/Runtime.cpp v8_snapshot_clang_x64/gen/v8/src/inspector/protocol/Runtime.h v8_snapshot_clang_x64/gen/v8/src/inspector/protocol/Schema.cpp v8_snapshot_clang_x64/gen/v8/src/inspector/protocol/Schema.h v8_snapshot_clang_x64/gen/v8/include/inspector/Debugger.h v8_snapshot_clang_x64/gen/v8/include/inspector/Runtime.h v8_snapshot_clang_x64/gen/v8/include/inspector/Schema.h
| /OE/lge/build/webosose/kirkstone/BUILD/work/x86_64-linux/mksnapshot-cross-x86_64/91.0.4472.114-11-r1/recipe-sysroot-native/usr/bin/python3-native/python3 ../../git/src/v8/third_party/inspector_protocol/code_generator.py --jinja_dir ../../git/src/third_party/ --output_base v8_snapshot_clang_x64/gen/v8/src/inspector --config ../../git/src/v8/src/inspector/inspector_protocol_config.json --inspector_protocol_dir //v8//third_party/inspector_protocol
| Traceback (most recent call last):
|   File "/OE/lge/build/webosose/kirkstone/BUILD/work/x86_64-linux/mksnapshot-cross-x86_64/91.0.4472.114-11-r1/build/Release/../../git/src/v8/third_party/inspector_protocol/code_generator.py", line 702, in <module>
|     main()
|   File "/OE/lge/build/webosose/kirkstone/BUILD/work/x86_64-linux/mksnapshot-cross-x86_64/91.0.4472.114-11-r1/build/Release/../../git/src/v8/third_party/inspector_protocol/code_generator.py", line 584, in main
|     jinja_env = initialize_jinja_env(jinja_dir, config.protocol.output, config)
|   File "/OE/lge/build/webosose/kirkstone/BUILD/work/x86_64-linux/mksnapshot-cross-x86_64/91.0.4472.114-11-r1/build/Release/../../git/src/v8/third_party/inspector_protocol/code_generator.py", line 190, in initialize_jinja_env
|     import jinja2
|   File "/OE/lge/build/webosose/kirkstone/BUILD/work/x86_64-linux/mksnapshot-cross-x86_64/91.0.4472.114-11-r1/git/src/third_party/jinja2/__init__.py", line 33, in <module>
|     from jinja2.environment import Environment, Template
|   File "/OE/lge/build/webosose/kirkstone/BUILD/work/x86_64-linux/mksnapshot-cross-x86_64/91.0.4472.114-11-r1/git/src/third_party/jinja2/environment.py", line 16, in <module>
|     from jinja2.defaults import BLOCK_START_STRING, \
|   File "/OE/lge/build/webosose/kirkstone/BUILD/work/x86_64-linux/mksnapshot-cross-x86_64/91.0.4472.114-11-r1/git/src/third_party/jinja2/defaults.py", line 32, in <module>
|     from jinja2.tests import TESTS as DEFAULT_TESTS
|   File "/OE/lge/build/webosose/kirkstone/BUILD/work/x86_64-linux/mksnapshot-cross-x86_64/91.0.4472.114-11-r1/git/src/third_party/jinja2/tests.py", line 13, in <module>
|     from collections import Mapping
| ImportError: cannot import name 'Mapping' from 'collections' (/OE/lge/build/webosose/kirkstone/BUILD/work/x86_64-linux/mksnapshot-cross-x86_64/91.0.4472.114-11-r1/recipe-sysroot-native/usr/lib/python3.10/collections/__init__.py)

The same issue is fixed upstream with new jinja2 in:
https://chromium-review.googlesource.com/c/chromium/src/+/3205163

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 src/third_party/jinja2/tests.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/third_party/jinja2/tests.py b/src/third_party/jinja2/tests.py
index 0adc3d4dbc..b14f85ff14 100644
--- a/src/third_party/jinja2/tests.py
+++ b/src/third_party/jinja2/tests.py
@@ -10,7 +10,7 @@
 """
 import operator
 import re
-from collections import Mapping
+from collections.abc import Mapping
 from jinja2.runtime import Undefined
 from jinja2._compat import text_type, string_types, integer_types
 import decimal
